File: Data Structures & Algorithms/jump-game-ii/submission-6.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int ret = 0;
        int lIndex = 0, rIndex = 0;

        while (rIndex < nums.size() - 1){
            int farthest = 0;
            for (int i = lIndex; i < rIndex + 1; i++){
                farthest = max(farthest, i + nums[i]);
            }
            lIndex = rIndex + 1;
            rIndex = farthest;
            ret++;
        }

        return ret;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l = 0

        ret = 0
        checked = set()
        for r in range(len(s)):
            while s[r] in checked:
                checked.remove(s[l])
                l += 1
            
            ret = max(ret, r - l + 1)
            checked.add(s[r])
        
        return ret